src/connector/src/sink/file_sink/opendal_sink.rs

```diff
@@ -292,25 +319,42 @@ impl OpenDalSinkWriter {
                     )?,
                 ));
             }
-            _ => unimplemented!(),
+            _ => {
+                self.sink_writer = Some(FileWriterEnum::FileWriter(object_writer));
+            }
         }
 
         Ok(())
     }
 
     async fn append_only(&mut self, chunk: StreamChunk) -> Result<()> {
-        let (data_chunk, _) = chunk.compact().into_parts();
-
         match self
             .sink_writer
             .as_mut()
             .ok_or_else(|| SinkError::File("Sink writer is not created.".to_string()))?
         {
             FileWriterEnum::ParquetFileWriter(w) => {
                 let batch =
-                    IcebergArrowConvert.to_record_batch(self.schema.clone(), &data_chunk)?;
+                    IcebergArrowConvert.to_record_batch(self.schema.clone(), chunk.data_chunk())?;
                 w.write(&batch).await?;
             }
+            FileWriterEnum::FileWriter(w) => {
+                let mut chunk_buf = BytesMut::new();
+
+                // write the json representations of the row(s) in current chunk to `chunk_buf`
+                for (op, row) in chunk.rows() {
+                    assert_eq!(op, Op::Insert, "expect all `op(s)` to be `Op::Insert`");
+                    // to prevent temporary string allocation,
+                    // so we directly write to `chunk_buf` implicitly via `write_fmt`.
+                    writeln!(
```

Comment:
in snowflake sink, there are no line breaks between lines, but the file sink has.(I think it's more friendly with user).

Reply:
Has it been tested to import snowflake?